### PR TITLE
8297209: Serial: Refactor GenCollectedHeap::full_process_roots

### PR DESCRIPTION
**Pull request by albertnetymk**
Simple change of replacing `MarkingCodeBlobClosure` with `CodeBlobToOopClosure` in the adjust-phase of Full GC.

Test: tier1-5

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8297209](https://bugs.openjdk.org/browse/JDK-8297209): Serial: Refactor GenCollectedHeap::full_process_roots

### Reviewers
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**)
 * [Ivan Walulya](https://openjdk.org/census#iwalulya) (@walulyai - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11210/head:pull/11210` \
`$ git checkout pull/11210`

Update a local copy of the PR: \
`$ git checkout pull/11210` \
`$ git pull https://git.openjdk.org/jdk pull/11210/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11210`

View PR using the GUI difftool: \
`$ git pr show -t 11210`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11210.diff">https://git.openjdk.org/jdk/pull/11210.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/shared/genCollectedHeap.cpp [hunk 1]
```diff
@@ -791,16 +791,19 @@ void GenCollectedHeap::process_roots(ScanningOption so,
...
                                           OopClosure* root_closure,
                                           CLDClosure* cld_closure) {
   // Called from either the marking phase or the adjust phase.
   const bool is_marking_phase = !is_adjust_phase;
 
-  MarkingCodeBlobClosure mark_code_closure(root_closure, is_adjust_phase, is_marking_phase);
-  CLDClosure* weak_cld_closure = only_strong_roots ? NULL : cld_closure;
-
-  process_roots(so, root_closure, cld_closure, weak_cld_closure, &mark_code_closure);
+  if (is_marking_phase) {
```
Comment (tschatzl) [inline src/hotspot/share/gc/shared/genCollectedHeap.cpp:799]: I do not think it is useful to explicitly create a local variable of the inverse of a parameter for its single use. That looks like the parameter is wrong.

I recommend either using `is_adjust_phase` directly here (potentially swapping the branches) or change the parameter of the method and the callers if you prefer to have `is_marking_phase` here.

**Review by tschatzl — CHANGES_REQUESTED** [review]

File: src/hotspot/share/gc/shared/genCollectedHeap.cpp (hunk 1 above)
Comment (albertnetymk) [reply src/hotspot/share/gc/shared/genCollectedHeap.cpp:799]: After this change, it becomes apparent this method should be split into two, for marking/adjust phase respectively, which I plan to address in the next PR.

Comment (tschatzl) [reply src/hotspot/share/gc/shared/genCollectedHeap.cpp:799]: I would not mind doing this in this PR - it is called `Refactor GenCollectedHeap::full_process_roots` after all, but I do not mind.

**Review by tschatzl — APPROVED** [review]

**Review by walulyai — APPROVED** [review]
Lgtm!

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**